cli/yamls.py

```diff
@@ -1,358 +1,217 @@
 """
 This module contains the tools for handling YAML files and tags.
 """
-
-import logging
-
 import os
 import re
-import sys
 import string
 
-import configure
 import yaml
 
 from cli import exceptions
 from cli import logger
 
 LOG = logger.LOG
 
-# Representer for Configuration object
-yaml.SafeDumper.add_representer(
-    configure.Configuration,
-    lambda dumper, value:
-    yaml.representer.BaseRepresenter.represent_mapping
-    (dumper, u'tag:yaml.org,2002:map', value))
 
+class Random(yaml.YAMLObject):
+    yaml_tag = u'!random'
+    yaml_dumper = yaml.SafeDumper
 
-def random_generator(size=32, chars=string.ascii_lowercase + string.digits):
-    import random
-
-    return ''.join(random.choice(chars) for _ in range(size))
+    @classmethod
+    def from_yaml(cls, loader, node):
+        import random
 
+        chars = string.ascii_lowercase + string.digits
+        return ''.join(random.choice(chars) for _ in range(int(node.value)))
 
-@configure.Configuration.add_constructor('join')
-def _join_constructor(loader, node):
-    seq = loader.construct_sequence(node)
-    return ''.join([str(i) for i in seq])
 
+class Placeholder(yaml.YAMLObject):
+    """ Raises 'IRPlaceholderException' when dumping Placeholder objects.
 
-@configure.Configuration.add_constructor('random')
-def _random_constructor(loader, node):
-    """
-    usage:
-        !random <length>
-    returns a random string of <length> characters
+    Objects created by 'from_yaml' method are automatically added to the
+    'placeholders_list' class variable so it'll be possible to add for each
+    object the path to the file where it stored.
     """
+    yaml_tag = u'!placeholder'
+    yaml_dumper = yaml.SafeDumper
 
-    num_chars = loader.construct_scalar(node)
-    return random_generator(int(num_chars))
+    # Refs for all Placeholder's objects
+    placeholders_list = []
 
+    def __init__(self, message):
+        self.message = message
+        self.file_path = None
 
-def _limit_chars(_string, length):
-    length = int(length)
-    if length < 0:
-        raise exceptions.IRException('length to crop should be int, not ' +
-                                     str(length))
+    @classmethod
+    def from_yaml(cls, loader, node):
```

Comment:
When loading a file using yaml.load, the value is replaced with Placeholder object..
When dumping the file using yaml.dump, if there are any Placeholder objects, an exception will be raised with the exact trace to the yaml file where this Placeholder came from.
This is important when having more than one placeholder and/or merging many yaml files